mindmeld/components/entity_resolver.py

```diff
@@ -42,60 +50,266 @@
     load_index,
     resolve_es_config_for_version,
 )
+from ._util import _is_module_available, _get_module_or_attr
+from .. import path
+from ..core import Entity
+from ..exceptions import EntityResolverConnectionError, EntityResolverError
 
 logger = logging.getLogger(__name__)
 
 
 class EntityResolver:
-    """An entity resolver is used to resolve entities in a given query to their canonical values
-    (usually linked to specific entries in a knowledge base).
-    """
 
-    # prefix for Elasticsearch indices used to store synonyms for entity resolution
-    ES_SYNONYM_INDEX_PREFIX = "synonym"
-    """The prefix of the ES index."""
-
-    def __init__(
-        self, app_path, resource_loader, entity_type, es_host=None, es_client=None
-    ):
-        """Initializes an entity resolver
+    @staticmethod
+    def _validate_resolver_name(name):
+        if name not in ENTITY_RESOLVER_MODEL_TYPES:
+            msg = "Expected 'model_type' in ENTITY_RESOLVER_CONFIG among {!r}"
+            raise Exception(msg.format(ENTITY_RESOLVER_MODEL_TYPES))
+        if name == "sbert_cosine_similarity" and not _is_module_available("sentence_transformers"):
+            raise ImportError(
+                "Must install the extra [bert] by running `pip install mindmeld[bert]`"
+                "to use the built in embbedder for entity resolution. See "
+                "https://www.mindmeld.com/docs/userguide/getting_started.html")
```

Comment:
Not sure how this link is useful for solving the error

Reply:
makes sense, removed it!